engine/src/main/java/org/terasology/entitySystem/event/internal/EventSystemImpl.java

```diff
@@ -241,6 +241,16 @@ private void addToComponentSpecificHandlers(Class<? extends Event> type, EventHa
                 }
             }
         }
+
+        if (0 == componentTypes.length) {
```

Comment:
Not a complaint at all but more an observation: we haven't really done [Yoda Conditions](https://en.wikipedia.org/wiki/Yoda_conditions) much before, but that might be more from just not having thought about them much. It seems like they have some decent pros. I'm unsure what I would go for if we try to think about what sort of convention we'd want to go with